src/parser/LiteralOrIriType.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023, University of Freiburg,
+//                 Chair of Algorithms and Data Structures.
+// Author: Benedikt Maria Beckermann <benedikt.beckermann@dagstuhl.de>
+
+#pragma once
+
+#include "IriType.h"
+#include "LiteralType.h"
+
+class LiteralOrIriType {
+ private:
+  using LiteralOrIriVariant = std::variant<LiteralType, IriType>;
+  LiteralOrIriVariant data;
+
+  // Returns contained IriType object if available, throws exception otherwise
+  IriType& getIriTypeObject();
+  // Returns contained LiteralType object if available, throws exception
+  // otherwise
+  LiteralType& getLiteralTypeObject();
```

Comment:
Just `getIri` and `getLiteral` but also add the `const& getBla() const;`
overloads.